docs/design/AgentRun-CLI-Design.md

```diff
@@ -0,0 +1,658 @@
+
+The behavior of the publish process depends on the **build** mode. It requires options:
+- **Image Repository URL and Credentials**
+    - **Local build mode**: The developer must explicitly provide the image repository URL, along with username and password, via CLI options.
+    - **Cloud build mode**: The image is already built and stored in the cloud image repository. The image location is automatically retrieved and merged into the metadata configuration file during the build process.
+- **Agent Version**
+    - A semantic version string (e.g., `v1.0.0`) used to register the agent.
+    - If publishing a new version, the version must be unique.
+    - If updating an existing version, the same version number can be reused.
+- **Additional Metadata**
+    - Any other fields required by AgentCube’s `create_or_update_agent` API, such as:
+        - Agent description
+        - Deployment region
+
+**Step 3.1**: Update Metadata Configuration File
+As part of the publish process, the CLI updates and merges deployment-related fields into the agent’s metadata configuration file to ensure consistency and traceability.
+
+**Fields updated include:**
+- **Agent version** – Used for version tracking and rollback
+- **Image repository URL** – Required for deployment and invocation
+- **Build mode and deployment region** – Confirmed or updated based on the current publish context
+- **Additional metadata** – Such as tags, and description, as required by AgentCube 
+
+**Step 3.2**: Tag and Push the Agent Image
+- The agent image is tagged using the specified version (e.g., `my-agent:v1.0.0`).
+- **Local build mode**:
+    - The CLI logs into the specified image repository using provided credentials
+    - The image is pushed to the repository.
+    - The final image URL is retrieved and merged into the metadata configuration file.
+- **Cloud build mode**:
+    - The image is already present in the cloud repository.
+    - The CLI verifies the image location and ensures it is correctly recorded in the metadata configuration file.
+
+**Step 3.3** Trigger Agent Registration via AgentCube API
+The CLI calls the `create_or_update_agent()` API to register or update the agent in AgentCube. This is an **asynchronous API**, allowing AgentCube to process the request in the background while the CLI monitors for completion.
+
+The API payload includes:
+- Agent metadata (name, version, description, tags, etc.)
+- Image repository URL
+- Runtime configuration (entrypoint, port, endpoint URI)
+- Deployment region and build mode
+
+**Step 3.4** Await AgentCube Response 
+The CLI waits for the asynchronous response from AgentCube, which confirms the result of the registration or update process. The response includes:
+- **agent_id** – Unique identifier assigned by AgentCube
+- **agent_endpoint** – The fully qualified HTTP endpoint for invoking the agent
```

Comment:
Accurately  we may not need the `fully qualified HTTP endpoint `. The agentRun system may not know its own entry address, say there is an loadbalancer in front of it. 

So to be accurate, this can be an relative url path.